### DISCUSSION TITLE
Strip comments from json

### DISCUSSION THREAD

**Discussion by avalanche1** [parcel-bundler/parcel#5821] [Q&A] [answered]
Is there a plugin to make Parcel v2 strip json comments?
Technically they are not allowed, but I like having them in my dev files.

**Comment by krnlde** [comment, marked as answer]
The syntax you'd like to have is called json5, I don't know if they want to add that.

**Reply by mischnic** [reply]
JSON files are parsed with json5 by default, so this should work anyway:

https://github.com/parcel-bundler/parcel/blob/92122c499d313c3db6cefbf1fdd190a7777e675f/packages/transformers/json/src/JSONTransformer.js#L14

**Reply by avalanche1** [reply]
Is that supposed to work for `manifest.json` in web extns?
Because it doesn't for Me.

**Reply by mischnic** [reply]
In that case: no. We use the `json-source-map` package there to also get code positions for generating errors with codeframes. And that one doesn't support json5

https://github.com/parcel-bundler/parcel/blob/92122c499d313c3db6cefbf1fdd190a7777e675f/packages/transformers/webmanifest/src/WebManifestTransformer.js#L39

**Reply by mischnic** [reply]
A transformer to convert json5 to json would look something like this:
```js
// @flow

import {Transformer} from '@parcel/plugin';
import json5 from 'json5';

export default (new Transformer({
  async transform({asset}) {
    asset.setCode(JSON.stringify(json5.parse(await asset.getCode())));
    return [asset];
  },
}): Transformer);
```
one downside would be that it also formats the JSON, so if you have a typo in the webmanifest, the Parcel error would point to the wrong line/column position.
